### ISSUE TITLE
newline-after-import has weird autofix for IIFE with leading semicolon

### ISSUE THREAD

**Issue by ricealexander** [benmosher/eslint-plugin-import#1759]
### Affected Code

I'm writing a file in which I require a file and then use an async IIFE to wrap my code.
```js
const composeTest = require('./lib/composeTest')

;(async () => {
  const tests = [
    await composeTest(/* arguments go here */),
  ]

  for (const test of tests) await test()
})()
```

This provides the error:
> Expected 1 empty line after require statement not followed by another require

And auto-fixes to this:
```js
const composeTest = require('./lib/composeTest')

;

(async () => {
  const tests = [
    await composeTest(/* arguments go here */),
  ]

  for (const test of tests) await test()
})()
```

Semicolons before IIFEs are required when relying on ASI which is why it is a common convention for developers who don't use them to place a semicolon immediately before them, as my first code sample shows.

### Configuration

My [full configuration](https://github.com/ricealexander/eslint-config-webtrinkets/blob/master/.eslintrc.js) is very large.

The following minimal configuration can reproduce this issue:
```
module.exports = {
  'parserOptions': {
    'ecmaVersion': 8,
  },

  'env': { 'node': true },
  'plugins': ['import'],

  'rules': {
    'semi': ['error', 'never'],
    'import/newline-after-import': 'error',
  },
}
```

### In Conclusion

I don't have any fixes in mind. I can move the semicolon to the end of the previous line and my code works and looks fine. I just wanted to provide the information that the auto-fix behaves this way in this situation.

**Comment by ljharb** [comment]
In general, I'd say IIFEs are obsolete now that modules exist, even async ones since TLA is shipping, and ASI hazards are going to keep getting worse as the language evolves, so you may want to reconsider that choice.

That said, I'm happy to accept a PR that improves the autofix here; but to set expectations, I'm not planning on working on it on my own.
